### ISSUE TITLE
Quantum fMRI analysis

### ISSUE THREAD

**Issue by robertloredo** [qiskit-advocate/qamp-fall-22#6]
### Description

Functional magnetic resonance imaging (fMRI) and magnetic resonance imaging (MRI) are techniques that allow soft tissue and blood to be represented as images. In this project we will be looking at ways to represent the fMRI information and determine innovative ways to analyze the information from an open-source data repository of controlled centers. Analysis can include determining whether a subject has a neuro-degenerative disease or identifying biomarkers of those diseases. 

In the 3 months we will work on how the information can best be represented in a quantum state and what interesting things can we extract from it using quantum computing principles. 

### Deliverables

1. A working circuit of the project running on both a simulator and physical quantum device 
2. A presentation describing the work done and outcomes
3. A draft for Qiskit tutorial/textbook

### Mentors details

- Mentor 1
  - Name: Robert Loredo
  - GitHub ID: @robertloredo
  - What they do: IBM Quantum ambassador worldwide lead

### Number of mentees

3

### Type of mentees

- Mentee 1
  - Required: 
    - Qiskit development skills
    - Experience on classic/quantum application development 
  - Nice to have:
    - Life science skills in medical imaging
- Mentee 2
  - Required: 
    - Qiskit development skills
    - Experience on classic/quantum application development 
  - Nice to have:
    - Medical informatics skills
- Mentee 3
  - Required: 
    - Qiskit development skills
    - Experience on classic/quantum application development 
  - Nice to have:
    - Image processing skills

**Comment by pdc-quantum** [comment]
Hi @robertloredo,
I am very interested in this project. I was reviewing the analysis of the fMRI information. I don't immediately see how to articulate classical and quantum computing in this project (I however see a few possible ways).

**Comment by jvscursulim** [comment]
Hi @robertloredo, I have an interest in joining this project. I have completed a course about image processing with Python in datacamp. In this course there is a module about biomedical image analysis.

**Comment by jodymburksphd** [comment]
Hi, @robertloredo I'd like to join the project as well.

**Comment by Tim-Li** [comment]
Hi, @robertloredo 
I am interested in this topic. I am working at the National Synchrotron Radiation Center in Taiwan. My main work is the application of deep learning in X-ray image (nano-CT) processing, including low-dose 3D tomographic reconstruction, denoise, and super-resolution. I am trying to apply quantum algorithms to X-ray images, so I hope to learn related knowledge through this topic.

**Comment by hsanthan** [comment]
Hi @robertloredo , I was recently working on a project to classify chest x-rays focusing on the transfer learning and data augmentation methods. This project seems to have a lot of learning opportunities. I am interested in joining the project. Thanks!

**Comment by Louisanity** [comment]
Hi @robertloredo , I am interested in this project. Recently I have done a medical imaging processing project using quantum transform learning with Qiskit. Also, I have 3 years of classic/quantum application development experience.

**Comment by robertloredo** [comment]
Hello everyone! 
@GemmaDawson will be creating two groups as we had a lot of interest in this topic. The two groups are as follows: 

Selection for Project #6 - Two groups:

**Group 1:**

- Tai Yue, Li (Group 1)
- Iulia Zidaro (Group 1) 
- Jody Burks (Group 1)
- Jose Victor (Group 1)

**Group 2:**

- Adnani Hinde (Group 2)
- Khadija Ech-challaouy (Group 2)
- Hamza Kamel Abdelsalam Attia Ahmed  (Group 2)
- KUAN-CHENG C (Group 2)


Stand by to receive an invitation on the GH repos that will be created for each group shortly. 

Could one person from each group please go ahead and create a group slack chat with those in your group (group 1 or group 2) to get us started as well? 

In the meantime, I am available from 8-10am, or 11:30-3pm ET on Wednesday. Please get with your group to find 30 mins so we can meet and please post the times here. Thank you and welcome aboard!! :) 

- Robert

**Comment by iuliazidaru** [comment]
@GemmaDawson

**Comment by jvscursulim** [comment]
@GemmaDawson

**Comment by GemmaDawson** [comment]
Please add your Checkpoint 1 presentation materials.

**Comment by jvscursulim** [comment]
Checkpoint 1 slides

[Project #6 - Quantum fMRI analysis.pptx](https://github.com/qiskit-advocate/qamp-fall-22/files/9747547/Project.6.-.Quantum.fMRI.analysis.pptx)

**Comment by HindeDS** [comment]
@GemmaDawson

**Comment by Tim-Li** [comment]
QAMP Fall 2022 – Cheek Point 2 – Description


The main target of this Project is to diagnose neurodegenerative disease. So, we focused on the second Checkpoint in diagnosing Autism. Autism Spectrum Disorder (ASD) affects approximately 1% of the population, causes social impairment, and is associated with lifelong disability. The lack of effective means of diagnosing ASD was attributed to the breadth and complexity of ASD symptoms and an incomplete understanding of brain functional connectivity.   
   
Functional MRI (fMRI) can indirectly measure neuronal activity using magnetic differences between oxygenated. And deoxygenated blood. In addition, functional connectivity can be captured when a patient is performing a specific task or when a patient is in a resting or task-negative state (resting-state fMRI).   
   
FMRI data are 4D image data composed of time series of 3D voxels. In recent years, machine learning and deep learning techniques have been widely used in fMRI datasets for ASD diagnosis. The advantage of machine learning is that feature extraction from fMRI data combined with known knowledge can increase diagnostic accuracy. However, machine learning is too subjective in extracting data features, limiting the accuracy of diagnosis on new data due to overfitting. It becomes complex to train machine learning models when the dataset has a greater number of features. The fewer features, the better the performance of the model.   
   
Deep learning can automatically and objectively extract features from data through convolutional neural networks. This feature has made convolutional neural networks successful in image vision. Therefore, using deep learning in ASD diagnosis is the current trend. However, deep learning techniques still have limitations. When the training data size is large, deep learning exceeds machine learning. Deep learning will have a lower or similar effect when the training data size is insufficient than machine learning.  
   
This situation is fatal in ASD diagnosis because fMRI data of ASD are not easy to obtain, and the data heterogeneity of ASD patients is high. Therefore, the method based on deep learning still faces a bottleneck in diagnosing ASD (the accuracy rate does not exceed 70%). Our team proposes to use a quantum machine learning-based approach to increase the accuracy of diagnosis in ASD. Recent studies have shown that quantum machine learning has advantages in CT medical image classification. It implies that quantum states may potentially express data characteristics.  
   
We used the Autism Brain Imaging Data Exchange (ABIDE) dataset to train the model, which supplies publicly available fMRI datasets for ASD (1112 datasets, of which 539 were from ASD individuals and 573 from typical controls). In addition, we use a general image preprocessing pipeline to integrate raw 4D fMRI images into 3D images for training.  
  
We propose three quantum machine learning strategies for comparison, which are 3D Quantum Convolutional Neural Network (3D QCNN), 3D Quantum Transfer Learning (3D QTL), and 3D Quantum Neural Network (3D QNN). 3D QCNN and 3D QTL combine convolutional neural network and quantum state, while 3D QNN combines quantum image representation and quantum state. We will implement these QML methods to compare and verify their effects.

**Comment by Tim-Li** [comment]
[IBM_QAMP_fall_22_checkpoint2.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9923553/IBM_QAMP_fall_22_checkpoint2.pdf)

**Comment by robertloredo** [comment]
This is great Team!! 

The PDF captures everything nicely. Have you posted the code for the various components? This will help us get two things started: 
1. Publication draft to either Qiskit Textbook or Blog
2. Prepare for next phase, interconnectivity towards multi-modality and/or toolbox. 

Let's sync up next week and discuss after I get back from conference. 

Great job! Really incredible progress!

**Comment by jvscursulim** [comment]
Here is the link for my GitHub repository where you all can find the implementations for NEQR, FRQI and QPIE circuits. There are also some notebooks about quantum transfer learning, QCNN and other machine learning stuff that can be useful for our project.

https://github.com/jvscursulim/qamp_fall22_project

**Comment by Tim-Li** [comment]
We are still working on 3D QCNN and 3D QTL(hybrid). After that, we will work on 3D QNN. We will write a draft when we finish all the parts.

**Comment by robertloredo** [comment]
Fantastic! Thank you John!

**Comment by Qcatty** [comment]
Hey @GemmaDawson

**Comment by hamzakamel1** [comment]
[FMRI project Check point 1.pptx.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9929895/FMRI.project.Check.point.1.pptx.pdf)
[IBM_QAMP_fall_22_checkpoint2.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9929896/IBM_QAMP_fall_22_checkpoint2.pdf)
[QAMP check point 2 - Description .pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9929902/QAMP.check.point.2.-.Description.pdf)

**Comment by hamzakamel1** [comment]
Hey @GemmaDawson

**Comment by hamzakamel1** [comment]
@GemmaDawson

**Comment by hamzakamel1** [comment]
https://user-images.githubusercontent.com/75124259/206894552-936a8fc1-92c1-4a37-a58c-708ce777c971.mp4

**Comment by hamzakamel1** [comment]
[Project #6 .pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/10201754/Project.6.pdf)

**Comment by GemmaDawson** [comment]
Congratulations on completing all the requirements for QAMP Fall 2022!! 🌟🌟🌟
